### ISSUE TITLE
Deprecate twisted.protocols.gps

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6810]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6810|
|Type|task|
|Created|2013-11-04 19:35:19Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-tprotocolsgps-6810|

(hawkowl note: all of twisted.protocols.gps should be deprecated because twisted.positioning is now here)

[twisted/protocols/gps/rockwell.py](twisted/protocols/gps/rockwell.py) was added 9 years ago (r4528) but never received any unit tests. Since then it was updated a few times (removing imports, adding license header) but unit tests were never added. [#6565](https://github.com/twisted/twisted/issues/6565) replaces usage of `reduce` in twisted core and also touches `rockwell.py`.

Because there are no existing tests, and interest for this protocol (from 1996) seems non-existent, we are going to deprecate it.

<details><summary>Searchable metadata</summary>

```
trac-id__6810 6810
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_deprecate_tprotocolsgps_6810 branches/deprecate-tprotocolsgps-6810
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1383593719000000 1383593719000000
changetime__1429703034357019 1429703034357019
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Also see [#3926](https://github.com/twisted/twisted/issues/3926) for `twisted.positioning`, "a better positioning framework" in the making.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [44515]) Branching to deprecate-tprotocolsgps-6810.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Changes made, builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes looks good to help.

Please fix the twistedchecker error and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44522]) Merge deprecate-tprotocolsgps-6810: Deprecate twisted.protocols.gps

Author: hawkowl
Reviewer: adiroiban
Fixes: [#6810](https://github.com/twisted/twisted/issues/6810)
